journal.py

```diff
@@ -86,35 +89,45 @@ def list_view(request):
 @view_config(route_name='detail', renderer='templates/detail.jinja2')
 def detail_view(request):
     entry = Entry.one(request.matchdict['id'])
-    return {'entry': entry}
+    html_text = markdown(entry.text, output_format='html5')
 
+    def my_highlight(matchobj):
+        return highlight(matchobj.group(0), PythonLexer(), HtmlFormatter())
 
-@view_config(route_name='create', renderer='templates/create.jinja2')
-def create_view(request):
-    return {}
+    pattern = r'(?<=<code>)[\s\S]*(?=<\/code>)'
+    html_text = re.sub(pattern, my_highlight, html_text)
+    return {
+        'entry': {
+            'id': entry.id,
+            'title': entry.title,
+            'text': html_text,
+            'created': entry.created
+        }
+    }
 
 
 @view_config(route_name='edit', renderer='templates/edit.jinja2')
 def edit_view(request):
+    if request.method == 'POST':
+        eid = request.matchdict['id']
+        title = request.params.get('title')
+        text = request.params.get('text')
+        Entry.modify(eid=eid, title=title, text=text)
+        return HTTPFound(request.route_url('home'))
+
     entry = Entry.one(request.matchdict['id'])
     return {'entry': entry}
```

Comment:
very good!

I'd suggest that since you are going to fetch the entry from the database anyway, you might make the 'modify' method an instance method, rather than a class method.  

As a question, is there anything that stops an anonymous user from editing an entry in your journal, if they should happen to know the ID of the post, and the correct url to go to?  As I see it, I should be able to load this page without any trouble, even if I'm not logged in.  Am I wrong?